source/adapters/level_zero/context.hpp

```diff
@@ -333,10 +334,38 @@ struct ur_context_handle_t_ : _ur_object {
         return WithProfiling ? &EventCaches[2] : &EventCaches[3];
       }
     }
+  };
+  auto getCounterBasedEventCache(bool UsingImmediateCmdList,
+                                 ur_device_handle_t Device) {
+    if (UsingImmediateCmdList) {
+      if (Device) {
+        auto EventCachesMap = &EventCachesDeviceMap[4];
+        if (EventCachesMap->find(Device) == EventCachesMap->end()) {
+          EventCaches.emplace_back();
+          EventCachesMap->insert(
+              std::make_pair(Device, EventCaches.size() - 1));
+        }
+        return &EventCaches[(*EventCachesMap)[Device]];
+      } else {
+        return &EventCaches[4];
+      }
+    } else {
+      if (Device) {
+        auto EventCachesMap = &EventCachesDeviceMap[5];
+        if (EventCachesMap->find(Device) == EventCachesMap->end()) {
+          EventCaches.emplace_back();
+          EventCachesMap->insert(
+              std::make_pair(Device, EventCaches.size() - 1));
+        }
+        return &EventCaches[(*EventCachesMap)[Device]];
+      } else {
+        return &EventCaches[5];
```

Comment:
why 5?
isn't EventCaches 6 in line 173 ?

Reply:
please make it more maintainable, we cannot have magic numbers scattered across code base.

Reply:
It's indexed at 0, 5 is the 6th slot.